packages/module/src/DataViewTh/DataViewTh.tsx

```diff
@@ -0,0 +1,282 @@
+import {
+  FC,
+  useState,
+  MouseEvent as ReactMouseEvent,
+  TouchEvent as ReactTouchEvent,
+  KeyboardEvent as ReactKeyboardEvent,
+  ReactNode,
+  useCallback,
+  useRef,
+  useEffect
+} from 'react';
+import { Th, ThProps } from '@patternfly/react-table';
+import { Button, getLanguageDirection } from '@patternfly/react-core';
+export interface DataViewThResizableProps {
+  /** Whether the column is resizable */
+  isResizable?: boolean;
+  /** Callback after the column is resized */
+  onResize?: (
+    event: ReactMouseEvent | MouseEvent | ReactKeyboardEvent | KeyboardEvent | TouchEvent,
+    width: number
+  ) => void;
+  /** Width of the column */
+  width?: number;
+  /** Minimum width of the column */
+  minWidth?: number;
+  /** Increment for keyboard navigation */
+  increment?: number;
+  /** Increment for keyboard navigation while shift is held */
+  shiftIncrement?: number;
+  /** Aria label for the resize button */
+  resizeButtonAriaLabel?: string;
+  /** Screenreader text for the column */
+  screenReaderText?: string;
+}
+export interface DataViewThProps {
+  /** Cell content */
+  content: ReactNode;
+  /** Resizable props */
+  resizableProps?: DataViewThResizableProps;
+  /** @hide Indicates whether the table has resizable columns */
+  hasResizableColumns?: boolean;
+  /** Props passed to Th */
+  props?: ThProps;
```

Comment:
Would we want to just have this named thProps like we do with PF components?

Reply:
I could rename it since this component isn't directly exposed, but `props` is what is exposed to the user so I wonder if it's worth being consistent with that?

Reply:
That's true. I don't feel strongly either way, so keeping it as-is would be fine

Reply:
I think I may for with `thProps` since it's an internal component, and more descriptive of what's being passed since the component is specifically for `th`.